dcrwallet.go

```diff
@@ -289,7 +290,22 @@ func rpcClientConnectLoop(legacyRPCServer *legacyrpc.Server, loader *wallet.Load
 				legacyRPCServer.SetChainServer(chainClient)
 			}
 			if cfg.EnableTicketBuyer {
-				startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				pm, err = startTicketPurchase(w, chainClient.Client, nil, cfg.tbCfg)
+				if err != nil {
+					log.Errorf("Unable to start ticket buyer: %v", err)
+					go func() {
+						loader.SetWaitFn(nil)
+					}()
+					return
+				}
+				pm.Start()
+				go func() {
```

Comment:
When `OpenWallet` grpc is called, the mutex is held by `OpenExistingWallet` at https://github.com/decred/dcrwallet/blob/master/wallet/loader.go#L194 and the function is called in `onLoaded` at https://github.com/decred/dcrwallet/blob/master/wallet/loader.go#L246. I think this was causing the deadlock since the callback was waiting for the lock which is held by `OpenExistingWallet`.